### ISSUE TITLE
Feature request: allow users to opt-in for duplicate names in `make_clean_names()`

### ISSUE THREAD

**Issue by JasonAizkalns** [sfirke/janitor#495]
## Feature requests

There are times when users may wish to _opt-in_ to allow duplicate names when using `make_clean_names()`. One approach would be having an additional argument: `make_clean_names(..., allow_dupes = FALSE, ...)`. Here is [some support](https://stackoverflow.com/q/62594765/2572423).

I believe a solution would be bypassing the final `while` loop in the `make_clean_names()` source (which was introduced in https://github.com/sfirke/janitor/pull/358):

```
...
cased_names <- snakecase::to_any_case(made_names, case = case, 
    sep_in = sep_in, transliterations = transliterations, 
    parsing_option = parsing_option, numerals = numerals, 
    ...)
while (any(duplicated(cased_names))) {
    dupe_count <- vapply(seq_along(cased_names), function(i) {
        sum(cased_names[i] == cased_names[1:i])
    }, 1L)
    cased_names[dupe_count > 1] <- paste(cased_names[dupe_count > 
        1], dupe_count[dupe_count > 1], sep = "_")
}
cased_names
```

So effectively:

```
...
cased_names <- snakecase::to_any_case(made_names, case = case, 
    sep_in = sep_in, transliterations = transliterations, 
    parsing_option = parsing_option, numerals = numerals, 
    ...)

if (!allow_dupes) {
  while (any(duplicated(cased_names))) {
    dupe_count <- vapply(seq_along(cased_names), function(i) {
      sum(cased_names[i] == cased_names[1:i])
    }, 1L)
    cased_names[dupe_count > 1] <- paste(cased_names[dupe_count > 
                                                       1], dupe_count[dupe_count > 1], sep = "_")
  }
}
cased_names
```

**Comment by sfirke** [comment]
That's a nicely done feature request. thank you.  I think I'm sold.  My reaction at first was, can't someone use `snakecake::to_any_case` ?  But looking at the code for `make_clean_names()` it does a lot of other things first that someone might want.

It's easy to implement, as you show, and there are already so many arguments to that function that what's another one at the end 😆 

Would you want to write a pull request that implements this new `allow_dupes` argument?  I could answer questions & give support.

**Comment by billdenney** [comment]
I agree that it makes sense as an option.  I agree with @sfirke that while `make_clean_names()` has a lot of arguments, a few more aren't a problem.  The main reason is that `make_clean_names()` is more of a detail-focused function, so having lots of controls makes sense.  (The opposite is true for `clean_names()` which is more general use, so fewer direct arguments are preferred.)

@JasonAizkalns Thanks in advance for the PR!

**Comment by sfirke** [comment]
If you both feel good about it I can get behind the breaking change. Thanks
both for the thoughtfulness here.

**Comment by billdenney** [comment]
@JasonAizkalns, thanks for proposing what feels like a great solution!  @sfirke, I do think that it's the better solution.

@JasonAizkalns, can you please revise the PR code to use the `unique_sep` argument with `make_clean_names()` and remove the while loop?  Some of the deduplication tests will also need to be revised, and please document it as a breaking change in the NEWS file.

**Comment by sfirke** [comment]
FYI we had some discussion about `unique_sep` vs the while-loop a couple of years ago, I am re-linking here in case any of those points feel salient: https://github.com/sfirke/janitor/issues/251#issuecomment-609408371

**Comment by JasonAizkalns** [comment]
Ah, @billdenney can you tell me more about your thoughts here: https://github.com/sfirke/janitor/issues/251#issuecomment-609408371 ?

**Comment by billdenney** [comment]
I don't remember the exact thoughts that I had.  Based on what I wrote, I'm guessing that I was concerned that duplicate names may still slip through, but I don't know why I had that concern.

I just did a quick check, and I don't think using it would be a problem.  I don't know if I checked it in the past and there was an issue or if it was concern about a possible bug if we didn't do a more extensive duplicate check.
